bots/succubus/succubus_main.lua

```diff
@@ -0,0 +1,584 @@
+
+local BotEcho, VerboseLog, BotLog = core.BotEcho, core.VerboseLog, core.BotLog
+local Clamp = core.Clamp
+
+local bottle = {}
+local illusionLib = object.illusionLib
+
+BotEcho(object:GetName()..' loading succubus_main...')
+
+----------------------------------------------------------
+--  			  bot constant definitions				--
+----------------------------------------------------------
+
+object.heroName = 'Hero_Succubis'
+
+--   item buy order. internal names  
+behaviorLib.StartingItems  = {"3 Item_MarkOfTheNovice", "Item_RunesOfTheBlight"}
+behaviorLib.LaneItems  = {"Item_Bottle", "Item_EnhancedMarchers"}
+behaviorLib.MidItems  = {"Item_PortalKey", "Item_Immunity", "Item_Summon 3"}
+behaviorLib.LateItems  = {"Item_Intelligence7", "Item_GrimoireOfPower"}
+--item_summon is puzzlebox; Item_Intelligence7 is master staff
+
+core.tLanePreferences = {Jungle = 0, Mid = 5, ShortSolo = 4, LongSolo = 4, ShortSupport = 3, LongSupport = 3, ShortCarry = 0, LongCarry = 0}
+
+-- Constants for skill usage
+
+object.mesmeUseBonus = 5
+object.holdUseBonus = 35
+object.heartacheUseBonus = 15
+
+object.mesmeUpBonus = 5
+object.holdUpBonus = 20
+object.heartacheUpBonus = 10
+
+object.holdThreshold = 60
+object.heartacheThreshold = 40
+object.mesmeThreshold = 50
+object.pkThreshold = 45
+
+------------------------------
+--	 skills			   --
+------------------------------
+-- skillbuild table, 0=smitten, 1=heartache, 2=mesme, 3=ult, 4=attri
+object.tSkills = {
+	1, 2, 1, 2, 1,
+	3, 1, 2, 2, 0, 
+	3, 0, 0, 0, 4,
+	3, 4, 4, 4, 4,
+	4, 4, 4, 4, 4,
+}
+function object:SkillBuild()
+	core.VerboseLog("skillbuild()")
+
+	local unitSelf = self.core.unitSelf
+	if  skills.smitten == nil then
+		skills.smitten = unitSelf:GetAbility(0)
+		skills.heartache = unitSelf:GetAbility(1)
+		skills.mesme = unitSelf:GetAbility(2)
+		skills.hold = unitSelf:GetAbility(3)
+	end
+	if unitSelf:GetAbilityPointsAvailable() <= 0 then
+		return
+	end
+
+	local nlev = unitSelf:GetLevel()
+	local nlevpts = unitSelf:GetAbilityPointsAvailable()
+	for i = nlev, nlev+nlevpts do
+		unitSelf:GetAbility( object.tSkills[i] ):LevelUp()
+	end
+end
+
+---------------------------------------------
+--	This adds puzzle box to the illusions  --
+---------------------------------------------
+
+function illusionLib.updateIllusions(botBrain)
+	illusionLib.tIllusions = {}
+	local tPossibleIllusions = core.tControllableUnits["AllUnits"]
+	if tPossibleIllusions ~= nil then
+		for nUID, unit in pairs(tPossibleIllusions) do
+			local sTypeName = unit:GetTypeName()
+			if sTypeName ~= "Pet_GroundFamiliar" and sTypeName ~= "Pet_FlyngCourier" then
+				tinsert(illusionLib.tIllusions, unit)
+			end
+		end
+	end
+end
+
+object.retreatCastThreshold = 55
+function behaviorLib.CustomRetreatExecute(botBrain)
+	local unitSelf = core.unitSelf
+	local mypos = unitSelf:GetPosition()
+
+	local lastRetreatUtil = behaviorLib.lastRetreatUtil
+
+	local missingHP = unitSelf:GetMaxHealth() - unitSelf:GetHealth()
```

Comment:
This is also only being used once, and would be more efficient not being defined.